tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -164,6 +164,21 @@ class Restrictions {
       ];
     }
 
+    var duplicateExtraClass =
+        config.extraClasses.cast<TypeDefinition?>().firstWhere(
+              (extraClass) => extraClass?.className == className,
+              orElse: () => null,
+            );
```

Comment:
Since collection is not a [dependency of serverpod_cli](https://github.com/serverpod/serverpod/blob/main/tools/serverpod_cli/pubspec.yaml), this seems to be the way to do it. Not sure if we want to consider adding the collection dependency like we do in the main serverpod package, which would contain the firstWhereOrNull helper? I would assume there would be more situations in which it would come in handy.

Reply:
It's probably not a bad thing to add the dependency here as well.
But keeping dependencies up to date is always a hassle. So if we are only interested in the `firstWhereOrNull` helper for lists, we might as well implement it as an extention ourselves and skip the dependency.